modules/mono/editor/GodotTools/GodotTools/Ides/Rider/RiderPathLocator.cs

```diff
@@ -1,185 +1,293 @@
+                RiderLocatorEnvironment.LogException(e);
             }
 
-            return Array.Empty<RiderInfo>();
+            return new RiderInfo[0];
         }
 
-        private static RiderInfo[] CollectAllRiderPathsLinux()
+        private RiderInfo[] CollectAllRiderPathsLinux()
         {
             var installInfos = new List<RiderInfo>();
-            string home = Environment.GetEnvironmentVariable("HOME");
+            var appsPath = GetAppsRootPathInToolbox();
+
+            installInfos.AddRange(CollectToolbox20Linux(appsPath, "*rider*", "bin/rider.sh"));
+            installInfos.AddRange(CollectToolbox20Linux(appsPath, "*fleet*", "bin/Fleet"));
+
+            var riderRootPath = Path.Combine(appsPath, "Rider");
+            installInfos.AddRange(CollectPathsFromToolbox(riderRootPath, "bin", "rider.sh", false)
+                .Select(a => new RiderInfo(this, a, true)).ToList());
+
+            var fleetRootPath = Path.Combine(appsPath, "Fleet");
+            installInfos.AddRange(CollectPathsFromToolbox(fleetRootPath, "bin", "Fleet", false)
+                .Select(a => new RiderInfo(this, a, true)).ToList());
+
+            var home = Environment.GetEnvironmentVariable("HOME");
             if (!string.IsNullOrEmpty(home))
             {
-                string toolboxRiderRootPath = GetToolboxBaseDir();
-                installInfos.AddRange(CollectPathsFromToolbox(toolboxRiderRootPath, "bin", "rider.sh", false)
-                  .Select(a => new RiderInfo(a, true)).ToList());
-
                 //$Home/.local/share/applications/jetbrains-rider.desktop
                 var shortcut = new FileInfo(Path.Combine(home, @".local/share/applications/jetbrains-rider.desktop"));
 
                 if (shortcut.Exists)
                 {
-                    string[] lines = File.ReadAllLines(shortcut.FullName);
-                    foreach (string line in lines)
+                    var lines = File.ReadAllLines(shortcut.FullName);
+                    foreach (var line in lines)
                     {
                         if (!line.StartsWith("Exec=\""))
                             continue;
-                        string path = line.Split('"').Where((item, index) => index == 1).SingleOrDefault();
+                        var path = line.Split('"').Where((_, index) => index == 1).SingleOrDefault();
                         if (string.IsNullOrEmpty(path))
                             continue;
 
                         if (installInfos.Any(a => a.Path == path)) // avoid adding similar build as from toolbox
                             continue;
-                        installInfos.Add(new RiderInfo(path, false));
+                        installInfos.Add(new RiderInfo(this, path, false));
                     }
                 }
             }
 
             // snap install
-            string snapInstallPath = "/snap/rider/current/bin/rider.sh";
+            var snapInstallPath = "/snap/rider/current/bin/rider.sh";
```

Comment:
nit-pick: I would prefer to undo all these changes from `string` to `var`. I think our code style guide prefers specifying the type in these cases.